src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,37 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    public static final DateTimeFormatter DATE_FORMAT = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+    private static final int DATE_INDEX = 0;
+    private static final int NAME_INDEX = 1;
+    private static final int HOURS_INDEX = 2;
+    private static final int SALARY_INDEX = 3;
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        int[] salary = new int[names.length];
+
+        for (String oneLine : data) {
+            String[] dataArray = oneLine.split(" ");
+            for (int i = 0; i < names.length; i++) {
+                if (!LocalDate.parse(dataArray[DATE_INDEX], DATE_FORMAT)
```

Comment:
let's create a separate variable for LocalDate.parse(dataArray[DATE_INDEX], DATE_FORMAT)